lib/login/accountCreation.dart

```diff
@@ -0,0 +1,199 @@
+  final TextEditingController cityController = TextEditingController();
+  final TextEditingController ageController = TextEditingController();
+  final TextEditingController genderController = TextEditingController();
+  final TextEditingController professionalController = TextEditingController();
+  final TextEditingController usernameController = TextEditingController();
+
+  bool visible = false;
+  Future<void> makePostRequest() async {
+    setState(() {
+      visible = true;
+    });
+    debugPrint('In postrequest');
+
+    final String email = widget.email;
+    final String password = 'HoldTheDoor';
+    final String deletedAt = 'null';
+    final String firstName = firstNameController.text;
+    final String lastName = lastNameController.text;
+    final String city = cityController.text;
+    final String age = ageController.text;
+    final String gender = genderController.text;
+    final String professional = professionalController.text;
+    final String username = usernameController.text;
+    final DateTime now = DateTime.now();
+    final String formatDate =
+        '${now.month.toString()}/${now.day.toString()}/${now.year.toString()}';
+
+    // API
+    const String address = 'http://localhost:9000/setUser';
+    final Map<String, String> headers = <String, String>{
+      'Content-type': 'application/json'
+    };
+
+    final dynamic userObject = User(age, city, formatDate, deletedAt, email,
+        firstName, gender, lastName, password, professional, username);
+
+    //userObject.age = age;
+
+    final http.Response call = await http.post(address,
+        headers: headers, body: json.encode(userObject.toJson()));
+
+    final int check = call.statusCode;
+    if (check == 200) {
+      debugPrint('call accepted');
+    } else {
+      debugPrint(check.toString());
+      throw Exception('Response failed to load');
+    }
+
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Welcome'),
+          actions: <Widget>[
+            FlatButton(
+              child: const Text('OK'),
+              onPressed: () {
+                Navigator.pushAndRemoveUntil(
+                    context,
+                    MaterialPageRoute<Widget>(
+                        builder: (BuildContext context) => const LoginPage()),
+                    (Route<void> route) => false);
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(title: const Text('Sign-Up Page')),
+        body: SingleChildScrollView(
+            child: Center(
+          child: Column(
+            children: <Widget>[
+              const Padding(
+                  padding: EdgeInsets.all(12.0),
+                  child: Text('Fill All Information in Form',
+                      style: TextStyle(fontSize: 22))),
+              Container(
```

Comment:
All these container's are typically the same except for a view a strings and a variable. What we can do is create a stateless function that returns a container that takes into two parameters and call it in this `children: <Widget> [ ... ]`

The function can look like this:
```
Widget getInputContainer(TextEditingController controllerType, String inputPlaceholder){ 
Container(
                  width: 280.0,
                  padding: const EdgeInsets.all(10.0),
                  child: TextField(
                    controller: controllerType,
                    autocorrect: true,
                    keyboardType: TextInputType.text,
                    textCapitalization: TextCapitalization.sentences,
                    decoration: const InputDecoration(hintText: inputPlaceholder),
                  )),
}
```

then in the Widget build() you can do this:

```
getInputContainer(firstNameController, 'Enter First Name Here'),
...
getInputContainer(userNameController, 'Create a username')

```

Reply:
also move this to a file called /components/login/InputTextContainer.dart 
If you can come up with a better filename then by all means go for it